reports/src/com/untangle/app/reports/ReportsApp.java

```diff
@@ -804,53 +804,59 @@ private String determineDbDriver()
      */
     private void writeCronFile()
     {
-        // write the cron file for nightly runs
-        String cronStr = "#!/bin/sh" + "\n" +
-            REPORTS_GENERATE_TABLES_SCRIPT + " | logger -t uvmreports" + "\n" +
-            REPORTS_CLEAN_TABLES_SCRIPT + " -d " + ReportsApp.dbDriver + " " + settings.getDbRetention() + " | logger -t uvmreports" + "\n" +
-            REPORTS_VACUUM_TABLES_SCRIPT + " | logger -t uvmreports" + "\n" +
-            REPORTS_GENERATE_FIXED_REPORTS_SCRIPT + " | logger -t uvmreports" + "\n";
-
-        if ( settings.getGoogleDriveUploadData() ) {
-            String dir = settings.getGoogleDriveDirectory();
-            if ( dir != null )
-                dir = " -d \"" + settings.getGoogleDriveDirectory() + "\"";
-            else
-                dir = "";
-            
-            cronStr += REPORTS_GOOGLE_DATA_BACKUP_SCRIPT + " " + dir + " | logger -t uvmreports" + "\n";
-        }
-        if ( settings.getGoogleDriveUploadCsv() ) {
-            String dir = settings.getGoogleDriveDirectory();
-            if ( dir != null )
-                dir = " -d \"" + settings.getGoogleDriveDirectory() + "\"";
-            else
-                dir = "";
-            
-            cronStr += REPORTS_GOOGLE_CSV_BACKUP_SCRIPT + " " + dir + " | logger -t uvmreports" + "\n";
-        }
+        // We only need to write this file if the database has daily retention set
+        if(settings.getDbRetention() > 0) {
+            // write the cron file for nightly runs
+            String cronStr = "#!/bin/sh" + "\n" +
+                REPORTS_GENERATE_TABLES_SCRIPT + " | logger -t uvmreports" + "\n" +
+                REPORTS_CLEAN_TABLES_SCRIPT + " -d " + ReportsApp.dbDriver + " " + settings.getDbRetention() + " | logger -t uvmreports" + "\n" +
```

Comment:
Only the "clean tables script" should be restricted by the daily retention.  All other scripts need to run daily.

Reply:
Done!